File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let answer = 0;
    const count = {};
    
    for (let i = 0; i < 10; i++) {
        count[discount[i]] = (count[discount[i]] | 0) + 1;
    }
    
    for (let i = 0; i <= discount.length - 10; i++) {
        if (want.every((item, index) => count[item] === number[index])) {
            answer++;
        }
        count[discount[i]] -= 1;
        count[discount[i+10]] = (count[discount[i+10]] | 0) + 1;
    }
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter

def solution(want, number, discount):
    answer = 0
    count = Counter(discount[:10])
    
    for i in range(len(discount) - 9):
        if all(count[item] == num for item, num in zip(want, number)):
            answer += 1
        if i + 10 < len(discount):
            count[discount[i]] -= 1
            count[discount[i+10]] += 1
    
    return answer
